azurerm/resource_arm_container_registry.go

```diff
@@ -203,6 +247,10 @@ func resourceArmContainerRegistryUpdate(d *schema.ResourceData, meta interface{}
 		}
 	}
 
+	if geoReplicationEnabled && strings.ToLower(sku) != strings.ToLower(string(containerregistry.Premium)) {
+		return fmt.Errorf("`georeplication_enabled` can only be enabled for a Premium Sku.")
+	}
```

Comment:
In the case where the user enables geo-replication then turns it off on the update, this will not currently remove the replications.

Reply:
update scenario with turning geo-replication off or updating to a lower Sku than premium is now implemented and covered by acceptance tests.